config/opal_setup_cc.m4

```diff
@@ -187,12 +187,26 @@ AC_DEFUN([OPAL_SETUP_CC],[
     if test "$GCC" = "yes"; then
         CFLAGS_orig="$CFLAGS"
 
+        # Note: Some versions of clang (at least >= 3.5 -- perhaps
+        # older versions, too?) will *warn* about -finline-functions,
+        # but still allow it.  This is very annoying, so check for
+        # that warning, too.
         CFLAGS="$CFLAGS_orig -finline-functions"
         add=
         AC_CACHE_CHECK([if $CC supports -finline-functions],
                    [ompi_cv_cc_finline_functions],
                    [AC_TRY_COMPILE([], [],
-                                   [ompi_cv_cc_finline_functions="yes"],
+                                   [ompi_cv_cc_finline_functions="yes"
+                                    if test -s conftest.err ; then
+                                        for i in unused 'not supported' ; do
```

Comment:
Can you put a comment here (or nearby) that shows what the format of Clang's message looks like?  That will help with future maintenance.